rpc_util: Change error message to indicate size after decompression

uds5501 opened the pull request:
Fixes #4761
 
RELEASE NOTES:
* client: clarify errors to indicate whether compressed or uncompressed messages exceeded size limits

uds5501 commented:
@menghanl @dfawley 
I have changed the error message as required in the issue, kindly let me know if any other change is required.

dfawley commented on rpc_util.go:
```diff
@@ -570,7 +570,7 @@ func (p *parser) recvMsg(maxReceiveMessageSize int) (pf payloadFormat, msg []byt
 	}
 	if int(length) > maxReceiveMessageSize {
-		return 0, nil, status.Errorf(codes.ResourceExhausted, "grpc: received message larger than max (%d vs. %d)", length, maxReceiveMessageSize)
+		return 0, nil, status.Errorf(codes.ResourceExhausted, "grpc: received message after decompression larger than max (%d vs. %d)", length, maxReceiveMessageSize)
```
This isn't correct.  This is the raw message before decompression.  `recvAndDecompress` calls this and decompresses this payload as a separate step.

dfawley replied:
Also note that we can't call this the "compressed" message, since we don't know whether it's compressed in the first place.  `pf` can tell us whether it's compressed if we can't come up with a generic enough message to convey the idea properly.

uds5501 replied:
@dfawley 
I have a small question, I was writing this fix after reading this comment here:

https://github.com/grpc/grpc-go/issues/4761#issuecomment-918817668

In this, the issue creator says that he is "compressing" an `8 MB` message to `4 MB` and then sending it over.
Now, in our logic, since we are decompressing it, later on, is it possible that your current logic and error used are absolutely right and the message was not compressed in the first place?

dfawley replied:
We check the message size twice.  Here (not yet decompressed) and here after decompressing:

https://github.com/grpc/grpc-go/blob/3b94303f375441714f9f4c488781e8928e000537/rpc_util.go#L721

And if a message isn't compressed, it seems like we might be checking the size twice, which we could clean up as part of this.  The issue is about clarifying whether it's the compressed message or the decompressed message size was over the limit (if the message is compressed).

uds5501 commented:
@dfawley Thank you for the review and pointers.

I have moved up this check so that we recheck the size once decompression is completed, also I have removed the message change from the first check itself.

dfawley commented:
Just noticed the vet error: `rpc_util.go:721:3: this value of size is never used (SA4006)`. You can remove the setting of `size` in the `else` to fix it.

uds5501 commented:
@dfawley Removed that, thanks for pointing out
